optimum/intel/generation/modeling.py

```diff
@@ -0,0 +1,345 @@
+        cache_dir: Optional[str] = None,
+        file_name: Optional[str] = WEIGHTS_NAME,
+        local_files_only: bool = False,
+        use_cache: bool = True,
+        **kwargs,
+    ):
+        to_torchscript = kwargs.pop("torchscript", False)
+        if config.torchscript:
+            task = cls.export_feature
+            model_kwargs = {
+                "revision": revision,
+                "use_auth_token": use_auth_token,
+                "cache_dir": cache_dir,
+                "subfolder": "",
+                "local_files_only": local_files_only,
+                "force_download": force_download,
+            }
+
+            model = TasksManager.get_model_from_task(task, "../../ipex/text-generation/model", **model_kwargs)
+            # Load the model from local directory
+            if os.path.isdir(model_id):
+                file_name = os.path.join(model_id, file_name)
+                model = cls.load_model(file_name)
+            # Download the model from the hub
+            else:
+                model_cache_path = hf_hub_download(
+                    repo_id=model_id,
+                    filename=file_name,
+                    use_auth_token=use_auth_token,
+                    revision=revision,
+                    cache_dir=cache_dir,
+                    force_download=force_download,
+                    local_files_only=local_files_only,
+                )
+                model = cls.load_model(model_cache_path)
+
+            # Only for IPEX, IPEX jit model need 2 iterations to convert model to int8 model
+            onnx_config_class = TasksManager.get_exporter_config_constructor(
+                model_type=config.model_type, exporter="onnx", task=cls.export_feature
+            )
+            onnx_config = onnx_config_class(config, use_past=use_cache)
+            model_inputs = onnx_config.generate_dummy_inputs(framework="pt")
+            for i in range(2):
+                model(**model_inputs)
+        elif not to_torchscript:
+            task = cls.export_feature
+            subfolder = kwargs.pop("subfolder", "")
+            model_kwargs = {
+                "revision": revision,
+                "use_auth_token": use_auth_token,
+                "cache_dir": cache_dir,
+                "subfolder": subfolder,
+                "local_files_only": local_files_only,
+                "force_download": force_download,
+            }
+            model = TasksManager.get_model_from_task(task, model_id, **model_kwargs)
+            model_inputs = None
+        else:
+            return cls._from_transformers(
+                model_id=model_id,
+                config=config,
+                use_auth_token=use_auth_token,
+                revision=revision,
+                force_download=force_download,
+                cache_dir=cache_dir,
+                local_files_only=local_files_only,
+                use_cache=use_cache,
+                **kwargs,
+            )

```

Comment:
When would this case happen ? `_from_transformers` should be called when `export=True`

```suggestion
```